github-runner-manager/tests/unit/openstack_cloud/test_openstack_cloud.py

```diff
@@ -241,3 +245,60 @@ def _mock_datetime_now(monkeypatch):
         "github_runner_manager.openstack_cloud.openstack_cloud.datetime", datetime_mock
     )
     return now
+
+
+def test_get_ssh_connection_success(openstack_cloud, monkeypatch):
+    """
+    arrange: Setup SSH connection to succeed.
+    act: Get SSH connection.
+    assert: No SSHError raised. No SSH connection errors.
+    """
+    mock_result = MagicMock(ok=True, stdout=_TEST_STRING)
+    mock_connection = MagicMock(run=MagicMock(return_value=mock_result))
+
+    def mock_ssh_connection(*_args, **_kwargs) -> MagicMock:
+        """Mock get_ssh_connection function.
+
+        Returns:
+            The mocked connection.
+        """
+        return mock_connection
+
+    monkeypatch.setattr(
+        "github_runner_manager.openstack_cloud.openstack_cloud.SSHConnection", mock_ssh_connection
+    )
+
+    mock_instance = MagicMock()
+    mock_instance.addresses = ["mock_ip"]
+    with openstack_cloud.get_ssh_connection(mock_instance):
+        pass
```

Comment:
Minor nit: I think for no-error raised tests, we could do `assert_<next_func>_called`  or cover them in the happy cases :)

Reply:
I assert that the returned connection is the mock_connection.
This would be testing whether the function is returning the right object.